src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/processorders/ProcessOrdersAboutToSubmitHandlerTest.java

```diff
@@ -62,6 +68,40 @@ void testCanHandle() {
         assertCanHandle(underTest, ABOUT_TO_SUBMIT, CONTESTED, PROCESS_ORDER);
     }
 
+    private static Stream<Arguments> provideInsertNewDocumentFromUnprocessedApprovedDocumentsToUploadHearingOrdersTestCase() {
+        return Stream.of(
+            Arguments.of(List.of()),
+            Arguments.of(List.of(
+                DirectionOrderCollection.builder().value(DirectionOrder.builder().uploadDraftDocument(TARGET_DOCUMENT_3).build()).build()
+            ))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideInsertNewDocumentFromUnprocessedApprovedDocumentsToUploadHearingOrdersTestCase")
+    void shouldInsertNewDocumentFromUnprocessedApprovedDocumentsToUploadHearingOrders(List<DirectionOrderCollection> uploadHearingOrder) {
+        FinremCallbackRequest finremCallbackRequest = FinremCallbackRequestFactory.from(FinremCaseData.builder()
+            .uploadHearingOrder(uploadHearingOrder)
+            .draftOrdersWrapper(DraftOrdersWrapper.builder()
+                .unprocessedApprovedDocuments(List.of(
+                    DirectionOrderCollection.builder().value(DirectionOrder.builder().originalDocument(TARGET_DOCUMENT_1)
+                        .uploadDraftDocument(TARGET_DOCUMENT_1).build()).build(),
+                    DirectionOrderCollection.builder().value(DirectionOrder.builder().originalDocument(TARGET_DOCUMENT_2)
+                        .uploadDraftDocument(TARGET_DOCUMENT_2).build()).build(),
+                    DirectionOrderCollection.builder().value(DirectionOrder.builder()
+                        .uploadDraftDocument(TARGET_DOCUMENT_3).build()).build()
+                ))
+                .build())
+            .build());
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> res =  underTest.handle(finremCallbackRequest, AUTH_TOKEN);
+
+        assertThat(res.getData().getUploadHearingOrder()).hasSize(uploadHearingOrder.size() + 1);
```

Comment:
I think this test would be more readable and more robust if you hard-coded the expected the value for size. If your expecting 3 then put 3.